sql/log_event.cc

```diff
@@ -1906,16 +1906,18 @@ Query_log_event::dummy_event(String *packet, ulong ev_offset,
 }
 
 /*
-  Replace an event (GTID event) with a BEGIN query event, to be compatible
+  Replace a Gtid_log_event with a BEGIN query event, to be compatible
   with an old slave.
 */
 int
 Query_log_event::begin_event(String *packet, ulong ev_offset,
+                             Format_description_log_event* fdev,
                              enum_binlog_checksum_alg checksum_alg)
 {
   uchar *p= (uchar *)packet->ptr() + ev_offset;
   uchar *q= p + LOG_EVENT_HEADER_LEN;
   size_t data_len= packet->length() - ev_offset;
+  Gtid_log_event original_event= Gtid_log_event(p, data_len, fdev);
```

Comment:
`Query_log_event::begin_event` occurs during pre-send, which is layers away from where the GTID event originates.
Persumably, this is because replication works by sending primaries’ already-serialized binary logs to replicas as relay logs.

The GTID event has a fixed-size format comprising two variable-sized sections, the thread_id being the very last optional entry.
This means, if I don’t leverage the existing constructor, I’ll have to duplicate its parsing code.

Let’s pretend that, instead of overwriting the buffer with a blank `BEGIN` query, we’re finally starting to *convert* the GTID event for real.